libpod/container_internal_linux.go

```diff
@@ -2368,3 +2385,30 @@ func (c *Container) checkFileExistsInRootfs(file string) (bool, error) {
 	}
 	return true, nil
 }
+
+// Creates and mounts an empty dir to mount secrets into, if it does not already exist
+func (c *Container) createSecretMountDir() error {
+	src := filepath.Join(c.state.RunDir, "/run/secrets")
+	_, err := os.Stat(src)
+	if os.IsNotExist(err) {
+		oldUmask := umask.Set(0)
+		defer umask.Set(oldUmask)
+
+		err = os.MkdirAll(src, 0644)
+		if err != nil {
+			return err
+		}
+		err = label.Relabel(src, c.config.MountLabel, false)
+		if err != nil {
+			return err
+		}
+		if err := os.Chown(src, c.RootUID(), c.RootGID()); err != nil {
```

Comment:
Non blocking nit: it would be nice to have one consistent way for error checks (`if err != nil` vs `if err := foo(); err != nil`).